### ISSUE TITLE
ivy errors when compiling docs in auctex

### ISSUE THREAD

**Issue by oracleyue** [abo-abo/swiper#1576]
In AucTeX, when we run `C-c C-c` (`Tex-command-master`) to compile LaTeX documents, I got the following error when using the newest Ivy&Counsel from elpa:

> sort: Wrong type argument: stringp, ("BibTeX" "bibtex %s" TeX-run-BibTeX nil t :help "Run BibTeX")

To reproduce it, you could use the following minimal config in your `.emacs`:
```lisp
;; packages
(package-initialize)

;; ivy and auctex
(ivy-mode 1)
(setq TeX-auto-save t)
(setq TeX-parse-self t)
```

This is quite critical, since many people use LaTeX almost everyday. I am curiously how they resolve this issue. Would there be a simple trick to solve it? Thanks!

**Comment by oracleyue** [comment]
Sorry I am not using the newest Ivy and Counsel. This is due to a mismatch of versions of these two packages or bugs just been fixed?.

**Comment by basil-conto** [comment]
> This is due to a mismatch of versions of Ivy and Counsel, or a bug just been fixed?

You were probably just missing the bug fix in #1568 for #1565 and #1567.
